src/gtc/gtir_to_oir.py

```diff
@@ -82,7 +82,10 @@ def visit_ParAssignStmt(
 
     def visit_FieldAccess(self, node: gtir.FieldAccess, **kwargs: Any) -> oir.FieldAccess:
         return oir.FieldAccess(
-            name=node.name, offset=node.offset, data_index=node.data_index, dtype=node.dtype
+            name=node.name,
+            offset=node.offset,
+            data_index=self.visit(node.data_index),
+            dtype=node.dtype,
```

Comment:
In fact this is a good example of the problem: It was not even properly lowering the expression nodes in the data index, so `oir` was left with some `gtir` nodes, and it didn't complain.